externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/SparkSQLEngine.scala

```diff
@@ -168,29 +173,39 @@ object SparkSQLEngine extends Logging {
 
   def main(args: Array[String]): Unit = {
     SignalRegister.registerLogger(logger)
-    var spark: SparkSession = null
-    try {
-      spark = createSpark()
+    setupConf()
+    val submitTime = kyuubiConf.getOption(KYUUBI_ENGINE_SUBMIT_TIME_KEY) match {
+      case Some(t) => t.toLong
+      case _ => System.currentTimeMillis()
+    }
+    val initTimeout = kyuubiConf.get(ENGINE_INIT_TIMEOUT)
+    if (System.currentTimeMillis() - submitTime > initTimeout) {
+      throw new KyuubiException(s"The maximum initialization time exceeded.")
```

Comment:
let's enrich the error msg here, mention
- The total time used
- the time submitted
- the ENGINE_INIT_TIMEOUT conf and its limited value